### PR TITLE
Release 1.4.2 prep

### PR DESCRIPTION
**Pull request by RedLeader962**
# Summary:

- CI related mod for multi-architecture build on NorLab build system
- Add support for Ubuntu Jammy
- Fix running test with cmake flag debug
- Update submodule version to latest release
- Various repository management fix

---

# Checklist:

### Code related

- [ ] I have made corresponding changes to the documentation 
      (i.e.: function, class, script header, README.md)
- [ ] I have commented hard-to-understand code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All tests pass locally with my changes 
      (Check [README.md #Contributing](https://github.com/norlab-ulaval/libpointmatcher/tree/develop#contributing)
      for local testing procedure using _libpointmatcher-build-system_)

### PR description related

- [x] I have included a quick summary of the changes
- [x] I have indicated the related issue's id with `# <issue-id>` if changes are of type `fix`
---

### Release PR related

- Only repository admins have the privilege to `push/merge` on the default branch (ie: `master`)
  and the `release` branch.

### CODE REVIEW

**Review by simonpierredeschenes — APPROVED** [review]
Looks good, great job!

File: build_system/docker-compose.libpointmatcher.yaml [hunk 1]
```diff
@@ -11,12 +11,12 @@ services:
     build:
       context: ..
       dockerfile: ${NBS_SUPERPROJECT_BUILD_SYSTEM_DIR:?err}/ubuntu/Dockerfile.libpointmatcher.compilation_test
-      no_cache: true
+      no_cache: false
       target: test-compilation-auto-path-resolution
       ## Mute 'platforms' while using arch virtualization with C++ build to prevent segmentation fault during lpm cmake install
```
[the remark this answers is not in the document]
Comment (RedLeader962) [reply build_system/docker-compose.libpointmatcher.yaml:16]: @simonpierredeschenes Good catch. Thanks for reading me back.

**Review by boxanm — COMMENTED** [review]
There are multiple places where the repo version is set to `1.4.0`. Is it correct?

**Comment by RedLeader962** [comment]
Yes those are doc or test related. The doc ones are for example like fetching a previous release. The test one could point to any existing release. We won't increment those at each release. Also I'm going to deploy semantic-release workflow for the next release so that version increments in files be automated. We won't waist time with this anymore and its not error-prone contrary to doing it manually.

**Comment by RedLeader962** [comment]
@boxanm By the way if its ok with you, I recommend we discard the Sonarcloud Quality Gate failure (Duplication of new code). At this moment Sonarcloud purpose for us is for providing code insight, not as rule. We will comeback to that issue eventually.

**Review by simonpierredeschenes — APPROVED** [review]

**Pull request merged**